chainer/training/updater.py

```diff
@@ -307,6 +308,10 @@ def update_core(self):
                 in_vars = variable.Variable(in_arrays)
                 losses.append(loss_func(in_vars))
 
+        # For _uninitialized_params
```

Comment:
I think this call to cleargrads(), just before the backward pass, is enough for all parameters. There is no need to clear them here: https://github.com/pfnet/chainer/blob/0ceb2b20d1cd7b16b90ceb8e78d55774e77777cb/chainer/training/updater.py#L292-L293 before the forward pass.